File: imstargg-frontend/src/model/Player.ts
import { SoloRankTier } from "./enums/SoloRankTier";

export interface Player {
  tag: string;
  name: string;
  nameColor: string;
  iconId: number;
  trophies: number;
  highestTrophies: number;
  soloRankTier: SoloRankTier | null;
  clubTag: string | null;
  updatedAt: Date;
}

File: core/core-domain/src/main/java/com/imstargg/core/domain/player/Player.java
package com.imstargg.core.domain.player;

import com.imstargg.core.domain.BrawlStarsId;
import com.imstargg.core.domain.BrawlStarsTag;
import com.imstargg.core.enums.PlayerStatus;
import com.imstargg.core.enums.SoloRankTier;
import jakarta.annotation.Nullable;

import java.time.OffsetDateTime;

public record Player(
        BrawlStarsTag tag,
        String name,
        String nameColor,
        BrawlStarsId iconId,
        int trophies,
        int highestTrophies,
        @Nullable SoloRankTier soloRankTier,
        @Nullable BrawlStarsTag clubTag,
        OffsetDateTime updatedAt,
        PlayerStatus status
) {
}
